### PR TITLE
README.md development help information

### PR DESCRIPTION
**Pull request by metsjeesus**
Just learning, i hope i got it right this time.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,15 +58,83 @@ For more information, read the [documentation](https://thelounge.github.io/docs/
 
 ## Development setup
 
-To run the app from source, just clone the code and run this in your terminal:
+
```
Comment (astorije) [inline README.md]: This is waaaay too much information.
I don't think people need so much to run from source, and it's fair to assume people have a reasonable knowledge of Git when they want to contribute. If they don't, https://git-scm.com/book/en/v2 is an extremely well-written resource, and its 3 first chapters are probably enough to be 80%-git-fluent, and cover 100% of what's needed to contribute to a project on GitHub.

File: README.md [hunk 2]
```diff
@@ -58,15 +58,83 @@ For more information, read the [documentation](https://thelounge.github.io/docs/
...
+    ```
+    npm start
+    ```
+
+5. Create new branch and switch into it, name should contain lounge issue ticket 
+    number, create one in https://github.com/thelounge/lounge/issues if missing
+
+    ```
+    git checkout -b XXX-feature-name
+    ```
```
Comment (astorije) [inline README.md]: This is too git-specific.
We don't require branch names to have the issue number in them (though maintainers are encouraged to include username+issue number when they commit to the main repo for organization purposes, but that's more common convention than requirement.
Actually, we don't require branches at all. It sure is a good idea if you are planning to open more than one PR on a project, but to each their own really.

File: README.md [hunk 3]
```diff
@@ -58,15 +58,83 @@ For more information, read the [documentation](https://thelounge.github.io/docs/
...
+    npm run build
+    npm run test
+    ```
+
+8. If test is passed, add changes and make commit
+
+    ```
+    git add -i 
+    git commit -m "#xxx - feature info"
+    ```
```
Comment (astorije) [inline README.md]: This is also git stuff and not The Lounge stuff, so it doesn't belong to our README, I think.

File: README.md [hunk 4]
```diff
@@ -58,15 +58,83 @@ For more information, read the [documentation](https://thelounge.github.io/docs/
...
+    git checkout -b XXX-feature-name
+    ``` 
+
+6. Edit code to add your feature.
+
+7. Build and test when its needed. You will have to run `npm run build` if you change or add anything in `client/js/libs` or `client/views`.
+ 
+    ```
+    npm run build
+    npm run test
```
Comment (astorije) [inline README.md]: `npm test` is a shorthand for `npm run test`.

File: README.md [hunk 5]
```diff
@@ -58,15 +58,83 @@ For more information, read the [documentation](https://thelounge.github.io/docs/
...
-npm start
-```
+pick fb524f5 This is commit 1
+s 2bd1943 This is commit 2
+s d911ebf This is commit 3
+```  
+    
+Merge commit messages, usually comment out whats not needed. 
+
+Finally upload to github repository with force flag  `git push origin XXX-feature-name --force`
```
Comment (astorije) [inline README.md]: L100 to L135 are git/GitHub-specific and should not be in the README.

**Comment by metsjeesus** [comment]
README.md is probably on wrong place to explain development procedure. I mainly made it to myself, because im very newbie on this git thing.  Make a wiki page or DEVHELP.md and README.md links to there? 

I will put local install method to development part too, because its only useful on testing.

Using `Git clone` instead of `npm install` is much more complicated and time consuming, because you need to install devDeps and compile sources too. If you just want to test something without editing source, `npm install` seems to do the trick just fine and you can delete it afterwards. But its for developers only.

**Comment by astorije** [comment]
Hey @metsjeesus, I'm hoping #621 clarifies the README for you! Thanks for your help and triggering me on this :-)
Closing this as there are clear overlaps. We can always re-open if necessary.

**Pull request closed without merging**